apps/angular/src/core/utils/compose-http-params.ts

```diff
@@ -0,0 +1,15 @@
+import { HttpParams } from '@angular/common/http';
+
+/**
+ * Compose fields from 'params' to new HttpParams object.
+ * @param params Query params.
+ */
+export function composeHttpParams(params: Readonly<Record<string, string>>): HttpParams {
+	let queryParams = new HttpParams();
+	for (const key in params) {
+		if (params[key] != null) {
+			queryParams = queryParams.set(key, params[key]);
+		}
+	}
+	return queryParams;
```

Comment:
In such cases it's better to use map/reduce.
```suggestion
	return Object.entries(params).reduce((queryParams, [key, value]) => {
		if (params[key] != null) {
			return queryParams.set(key, value);
		}
		return queryParams;
	}, new HttpParams());
```